Makefile

```diff
@@ -28,33 +28,46 @@ compile:
+	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PERL5LIB' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 	echo 'perl $$script_dir/../$(LIB_DIR)/$(SERVICE_CAPS)/$(SERVICE_CAPS)Server.pm $$1 $$2 $$3' >> $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 	chmod +x $(LBIN_DIR)/$(EXECUTABLE_SCRIPT_NAME)
 
 build-startup-script:
 	mkdir -p $(LBIN_DIR)
 	echo '#!/bin/bash' > $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
-	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'if [ -L $$0 ] ; then' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$(readlink $$0)")"; pwd -P)' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'else' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'script_dir=$$(cd "$$(dirname "$$0")"; pwd -P)' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'fi' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	echo 'export KB_DEPLOYMENT_CONFIG=$$script_dir/../deploy.cfg' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
-	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PATH:$$PERL5LIB' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
+	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PERL5LIB' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	echo 'plackup $$script_dir/../$(LIB_DIR)/$(SERVICE_CAPS).psgi' >> $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 	chmod +x $(SCRIPTS_DIR)/$(STARTUP_SCRIPT_NAME)
 
 build-test-script:
 	echo '#!/bin/bash' > $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'export KB_DEPLOYMENT_CONFIG=$$script_dir/../deploy.cfg' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'echo "Running $$0 with args $$@"' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'if [ -L $$0 ] ; then' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'test_dir=$$(cd "$$(dirname "$$(readlink $$0)")"; pwd -P) # for symbolic link' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'else' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'test_dir=$$(cd "$$(dirname "$$0")"; pwd -P) # for normal file' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'fi' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'base_dir=$$(cd $$test_dir && cd .. && pwd);' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'export KB_DEPLOYMENT_CONFIG=$$base_dir/deploy.cfg' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
 	echo 'export KB_AUTH_TOKEN=`cat /kb/module/work/token`' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PATH:$$PERL5LIB' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'cd $$script_dir/../$(TEST_DIR)' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo "perl -e 'opendir my \$$dh, \".\"; my @l = grep { /\\\\.pl\$$/ } readdir \$$dh; foreach my \$$s (@l) { print(\"Running \".\$$s.\"\\n\"); system \"perl\", \$$s; }'" >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'export PERL5LIB=$$base_dir/$(LIB_DIR):$$PERL5LIB' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'cd $$base_dir' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'prove -I $$test_dir -lvrm $$test_dir' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
```

Comment:
Perl tests are now run with the `prove` command; any test failures are reported at the end of the test run in the following format:
```

Test Summary Report
-------------------
/kb/module/test/annotation_utils.t        (Wstat: 256 Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 1
/kb/module/test/metagenome_utils.t        (Wstat: 256 Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 1
/kb/module/test/my_assembly_test.t        (Wstat: 512 Tests: 7 Failed: 2)
  Failed tests:  2, 4
  Non-zero exit status: 2
Files=5, Tests=48, 1008 wallclock secs ( 0.12 usr  0.06 sys + 195.20 cusr 20.25 csys = 215.63 CPU)
Result: FAIL

```